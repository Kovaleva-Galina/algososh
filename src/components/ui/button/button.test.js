import renderer  from 'react-test-renderer';

import { Button } from './button';

it('Корректная отрисовка кнопки с текстом', () => {
    const container = renderer.create(<Button text="Рассчитать" />).toJSON();
    expect(container).toMatchSnapshot();
}); 

it('Корректная отрисовка кнопки без текста', () => {
    const container = renderer.create(<Button />).toJSON();
    expect(container).toMatchSnapshot();
}); 

it('Корректная отрисовка заблокированной кнопки', () => {
    const container = renderer.create(<Button disabled />).toJSON();
    expect(container).toMatchSnapshot();
}); 

it('Корректная отрисовка кнопки с индикацией загрузки', () => {
    const container = renderer.create(<Button  isLoader = 'false' />).toJSON();
    expect(container).toMatchSnapshot();
}); 