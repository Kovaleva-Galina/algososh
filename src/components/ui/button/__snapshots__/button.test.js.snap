// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Корректная отрисовка заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button  false "
  disabled={true}
  type="button"
>
  <p
    className="text "
  />
</button>
`;

exports[`Корректная отрисовка кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button  false "
  type="button"
>
  <p
    className="text "
  />
</button>
`;

exports[`Корректная отрисовка кнопки с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button  loader "
  disabled="false"
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Корректная отрисовка кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button  false "
  type="button"
>
  <p
    className="text "
  >
    Рассчитать
  </p>
</button>
`;
